internal/db/resources.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 Canonical.
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// MULTI_TABLES_RAW_SQL contains the raw query fetching entities from multiple tables, with their respective entity parents.
+const MULTI_TABLES_RAW_SQL = `
+(
+	SELECT 'application_offer' AS type, 
+			application_offers.uuid AS id, 
+			application_offers.name AS name, 
+			models.uuid AS parent_id,
+			models.name AS parent_name,
+			'model' AS parent_type
+	FROM application_offers
+	JOIN models ON application_offers.model_id = models.id
```

Comment:
Why do we need the parent objects? Does the rebac-admin spec require that?

Reply:
yes, you can find in the issue on Jira the entities who need the parent as well https://warthogs.atlassian.net/browse/CSS-9402